src/app/header/header.component.ts

```diff
@@ -27,7 +28,8 @@ export class HeaderComponent implements OnInit {
 
   constructor(
     private store: Store<AppState>,
-    private windowService: HostWindowService
+    private windowService: HostWindowService,
+    public translate: TranslateService
```

Comment:
don't think this is needed here anymore

Reply:
In the list of languages to display I would use the labels in their respective language. There is no mechanism in ngx-translate for this kind of having a locale and translate a certain part with this locale rather than the current one, so I've done:
https://github.com/cjuergen/dspace-angular/blob/language-switch/resources/i18n/en.json#L277
Looks a bit odd, but there is no way to put comments in json.